ls1intum/Artemis
# `General`: Add copy button to LTI configuration

### Checklist
#### General
#### Client

### Motivation and Context
1. It can be annoying to have to copy all the LTI configuration fields manually.
2. When an LTI user is initialized, they are presented with a checkbox explaining their account was created and a password was created as well. They should save the password or they will have to reset it. It's easy for users to click ok without actually reading the dialog.
3. The LTI configuration page was showing a wrong breadcrumb ("Programming Exercises")

### Description
1. Added a reusable copy-icon-button and used it in all the LTI configuration fields for which a copy button is useful.
2. Added a checkbox so that users read the LTI initialization 
3. Fixed the breadcrumb to "LTI Configuration"

### Steps for Testing
Prerequisites:
- 1 Instructor
- Login to moodle https://moodle.ase.in.tum.de/

Test copy button and breadcrumb fix

1. Log in to Artemis
2. Navigate/create online course with exercise, navigate to exercise page and then LTI configuration
3. Verify that the copy icon works as expected
4. Verify that the breadcrumb has been fixed

Test new checkbox

1. Launch an LTI exercise as a new user
2. Verify that the checkbox is present in the initialization dialog and that the user can only close the dialog when the checkbox is clicked

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots

####  LTI initialization checkbox

##### Before
![ltiUserInitialization](https://user-images.githubusercontent.com/11006274/196517846-186e9ba7-a44b-4cad-99e1-2117e8f54b9a.png)

##### After

![ltiUserInitAfter](https://user-images.githubusercontent.com/11006274/196519531-f6d023f7-d84d-40e1-bb9b-aec8979da575.png)

#### Copy buttons and breadcrumb

##### Before
![ltiConfigurationBefore](https://user-images.githubusercontent.com/11006274/196517816-355ab6cb-36a0-4bbc-8a91-a91556133adf.png)

##### After
![afterWithCopyButtons](https://user-images.githubusercontent.com/11006274/196517855-f92de7b5-9f11-40f0-b69d-e198a0011238.png)
